packages/nameguard-react/src/components/Report/ResultCodeIcon.tsx

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import { Tooltip } from "../Tooltip/Tooltip";
+import { CheckResultCode, Rating } from "@namehash/nameguard";
+import { CheckResultCodePassIcon } from "../icons/CheckResultCodePassIcon";
+import { CheckResultCodeWarnIcon } from "../icons/CheckResultCodeWarnIcon";
+import { CheckResultCodeAlertIcon } from "../icons/CheckResultCodeAlertIcon";
+import { CheckResultCodeInfoIcon } from "../icons/CheckResultCodeInfoIcon";
+import { CheckResultCodeSkipIcon } from "../icons/CheckResultCodeSkipIcon";
+
+function text(code: CheckResultCode | Rating) {
```

Comment:
I'm strongly against using the type system this way. Let's please totally separate `CheckResultCode` from `Rating`.

The test if code is good or not requires a higher standard than just asking the question: Does it work? We also need to pass the test of: Is it maintainable? Is it easy for other developers to understand and use correctly without bugs being created?

For me, this fails those tests.

Also important: a `CheckResultCode` is conceptually different from a `Rating`. We should avoid smashing these ideas together as if they are the same thing. They are not.

Reply:
https://github.com/namehash/nameguard/pull/256#discussion_r1600551012